modules/diskq/qdisk.c

```diff
@@ -561,6 +561,21 @@ qdisk_pop_head(QDisk *self, GString *record)
   return TRUE;
 }
 
+gboolean
+qdisk_skip_head(QDisk *self)
```

Comment:
`LogQueueDisk` is the proper interface for the data structure, where `push`, `pop`, `ack`, `rewind` are available, `QDisk` is the lowest level of this module, where the interface communicates the whole internal representation, reader, writer heads, etc; but I agree, skip_head can be confusing.

We already have a method in QDisk, which is called `qdisk_skip_record()`.
@alltilla I think you've re-implemented `qdisk_skip_record()` here :)

Reply:
@MrAnno true, `qdisk_skip_record()` does a lot of similar things, but it does not do these:
```
  self->hdr->length--;

  if (!self->options->reliable)
    {
      self->hdr->backlog_head = self->hdr->read_head;

      g_assert(self->hdr->backlog_len == 0);
      if (!self->options->read_only)
        {
          qdisk_reset_file_if_empty(self);
        }
    }
```
I would not merge those two functions. We can use `_try_reading_record_length()` in `qdisk_skip_record()` though.

Reply:
@Kokan I am okay with renaming it to `qdisk_remove_head()`. Thanks!

Reply:
Can we push @MrAnno's comment?
We could at least refactor ye old `qdisk_skip_record` to use some the new internal functions (try_read_record_length, etc). Then we could also remove `qdisk_read` then entirely.

Reply:
We talked with @MrAnno IRL about the `qdisk_skip_record()` and `qdisk_remove_head()` question.
We agreed, that they do similar things, and they should be refactored into some common function, but seeing the differences, and the necessary call-site changes for that, we postponed the refactor. The `QDisk` code has multiple improvement possibilities, this is just one of them.

Reply:
@gaborznagy Sorry, I did not see your comment. Unfortunately `_try_reading_record_length()` has a side-effect, which moves the `read_head`, so it is entirely `read_head` specific. Maybe `_read_record_length_from_disk()` can be used if you insist. It can be generalized, so it reads from a given position.

Reply:
Okay, you already improved a lot of things.